### PR TITLE
pay logic -> only API essentials (bug 1130477)

### PR DESCRIPTION
**Pull request by kumar303**
@muffinresearch r? @andymckay r?

I manually tested purchase and PIN reset locally (with Zippy only). I also migrated old tests so that they hit the API instead of the old payment page.

### CODE REVIEW

File: webpay/urls.py [hunk 1]
```diff
@@ -10,16 +12,18 @@
...
+    '',
     (r'^mozpay/auth/', include('webpay.auth.urls')),
     (r'^mozpay/bango/', include('webpay.bango.urls')),
     (r'^mozpay/provider/', include('webpay.provider.urls')),
     (r'^mozpay/services/', include('webpay.services.urls')),
-    (r'^mozpay/', include('webpay.pay.urls'))
+    url(r'^mozpay/$', index, name='index'),
 )
 
 if settings.SPA_ENABLE_URLS:
```
Comment (muffinresearch) [inline webpay/urls.py]: These conditionals can all be killed now too. Unless you're saving that for later?

File: settings_test.py [hunk 2]
```diff
@@ -56,7 +56,7 @@
 
 CACHE_PREFIX = 'webpay:test'
 
-SPA_ENABLE = False
+SPA_ENABLE = True
```
Comment (muffinresearch) [inline settings_test.py]: As mentioned these can die too - but later if that makes it easier.

**Comment by kumar303** [comment]
oh yeah! I'll see how crazy it is to kill off SPA_ENABLE everywhere.

File: webpay/pay/urls.py [hunk 3]
```diff
@@ -1,39 +1,18 @@
...
-    url(r'^callback_success_url$', views.callback_success_url,
+    url(r'^callback_success_url$', callback_success_url,
         name='pay.callback_success_url'),
-    url(r'^callback_error_url$', views.callback_error_url,
+    url(r'^callback_error_url$', callback_error_url,
         name='pay.callback_error_url'),
 )
-
-
-if settings.DEBUG:
```
Comment (muffinresearch) [inline webpay/pay/urls.py]: I think the error views are still relevant to some degree. This enables testing they look ok. (The styles are coming from spartacus static iirc).

File: webpay/base/utils.py [hunk 4]
```diff
@@ -68,6 +68,7 @@ def system_error(request, **kw):
 
 
 def custom_error(request, user_message, code=msg.UNEXPECTED_ERROR, status=400):
+    from webpay.base.helpers import fxa_auth_info
```
Comment (andymckay) [inline webpay/base/utils.py]: boo

File: webpay/pay/views.py [hunk 5]
```diff
@@ -181,321 +181,6 @@ def configure_transaction(request, data=None):
...
-            return http.HttpResponseRedirect(
-                '{0}?next={1}'.format(reverse('pay.bounce'), redirect_url)
-            )
-
-    # If the buyer closed the trusted UI during reset flow, we want to unset
-    # the reset pin flag. They can hit the forgot pin button if they still
-    # don't remember their pin.
-    if sess.get('uuid_needs_pin_reset'):
-        try:
-            solitude.set_needs_pin_reset(sess['uuid'], False)
```
Comment (andymckay) [inline webpay/pay/views.py]: I don't think `solitude.set_needs_pin_reset` is used anywhere now, lets remove that?

File: webpay/pay/views.py [hunk 6]
```diff
@@ -181,321 +181,6 @@ def configure_transaction(request, data=None):
...
-    have_jwt = bool(request.GET.get('req'))
-
-    log.info('starting from JWT? {have_jwt}'.format(have_jwt=have_jwt))
-    if have_jwt:
-        # If it returns a response there was likely
-        # an error and we should return it.
-        res = process_pay_req(request)
-        if isinstance(res, http.HttpResponse):
-            return res
-    elif settings.TEST_PIN_UI:
```
Comment (andymckay) [inline webpay/pay/views.py]: `TEST_PIN_UI` is still in `settings_test.py`

File: webpay/pay/views.py [hunk 7]
```diff
@@ -181,321 +181,6 @@ def configure_transaction(request, data=None):
...
-            )
-
-    # If the buyer closed the trusted UI during reset flow, we want to unset
-    # the reset pin flag. They can hit the forgot pin button if they still
-    # don't remember their pin.
-    if sess.get('uuid_needs_pin_reset'):
-        try:
-            solitude.set_needs_pin_reset(sess['uuid'], False)
-        except ResourceModified:
-            return system_error(request, code=msg.RESOURCE_MODIFIED)
```
Comment (andymckay) [inline webpay/pay/views.py]: I think this the only place `RESOURCE_MODIFIED` is raised like this.

**Comment by andymckay** [comment]
r+wc thanks for doing that, more deletions to come or is this most of it?

File: webpay/base/utils.py (hunk 4 above)
Comment (kumar303) [inline webpay/base/utils.py]: heh. Yeah. Circular imports :facepunch:

File: webpay/pay/urls.py (hunk 3 above)
Comment (muffinresearch) [inline webpay/pay/urls.py]: I think there might be some error templating (I thought I saw being deleting) but that might have been an old diff.

File: webpay/pay/views.py (hunk 6 above)
Comment (kumar303) [inline webpay/pay/views.py]: fixed

File: webpay/pay/urls.py (hunk 3 above)
Comment (kumar303) [inline webpay/pay/urls.py]: ah, thanks, I'll put them back. I wasn't sure what that was for tbh.

File: webpay/pay/views.py (hunk 7 above)
Comment (kumar303) [inline webpay/pay/views.py]: some solitude responses [still return 412s](https://github.com/mozilla/solitude/search?utf8=%E2%9C%93&q=412&type=Code) so we have mock tests that simulate those responses in webpay. These still look to be relevant.

Comment (andymckay) [inline webpay/pay/views.py]: fair enough

**Comment by kumar303** [comment]
We still need to clean up persona auth: https://bugzilla.mozilla.org/show_bug.cgi?id=1130465

File: webpay/pay/views.py (hunk 5 above)
Comment (kumar303) [inline webpay/pay/views.py]: Good idea, I zapped it along with ``unset_was_locked``

**Pull request merged**